pgmpy/readwrite/XMLBIF.py

```diff
@@ -176,3 +177,202 @@ def get_model(self):
 
         model.add_cpds(*tabular_cpds)
         return model
+
+
+class XMLBIFWriter:
+    """
+    Base class for writing XMLBIF network file format.
+    """
+    def __init__(self, model, encoding='utf-8', prettyprint=True):
+        """
+        Initialise a XMLBIFWriter object.
+
+        Parameters
+        ----------
+        model: Model to write
+        encoding: String(optional)
+            Encoding for text data
+        prettyprint: Bool(optional)
+            Indentation in output XML if true
+
+        Examples
+        -------
+        >>> writer = XMLBIFWriter(model)
+        """
+        if not isinstance(model, BayesianModel):
+            raise TypeError("model must an instance of BayesianModel")
+        self.model = model
+
+        self.encoding = encoding
+        self.prettyprint = prettyprint
+
+        self.xml = etree.Element("BIF", attrib={'version': '0.3'})
+        self.network = etree.SubElement(self.xml, 'NETWORK')
+        try:
+            etree.SubElement(self.network, 'NAME').text = self.model['network_name']
+        except KeyError:
+            pass
+
+        self.variables = self.add_variables()
+        self.states = self.add_states()
+        self.properties = self.add_properties()
+        self.definition = self.add_definition()
+        self.tables = self.add_cpd()
+
+    def __str__(self):
+        """
+        Return the XML as string.
+        """
+        if self.prettyprint:
+            self.indent(self.xml)
+        return etree.tostring(self.xml, encoding=self.encoding)
+
+    def indent(self, elem, level=0):
+        """
+        Inplace prettyprint formatter.
+        """
+        i = "\n" + level*"  "
+        if len(elem):
+            if not elem.text or not elem.text.strip():
+                elem.text = i + "  "
+            if not elem.tail or not elem.tail.strip():
+                elem.tail = i
+            for elem in elem:
+                self.indent(elem, level+1)
+            if not elem.tail or not elem.tail.strip():
+                elem.tail = i
+        else:
+            if level and (not elem.tail or not elem.tail.strip()):
+                elem.tail = i
+
+    def add_variables(self):
+        """
+        Add variables to XMLBIF
+
+        Return
+        ---------------
+        xml containing variables tag
+
+        Examples
+        -------
+        >>> writer = XMLBIFWriter(model)
+        >>> writer.get_variables()
+        """
+        variables = self.model.nodes()
+        variable_tag = {}
+        for var in sorted(variables):
+            variable_tag[var] = etree.SubElement(self.network, "VARIABLE", attrib={'TYPE': 'nature'})
+            etree.SubElement(variable_tag[var], "NAME").text = var
+        return variable_tag
+
+    def add_states(self):
+        """
+        Add outcome to variables of XMLBIF
+
+        Return
+        ---------------
+        xml containing outcome tag
```

Comment:
@vivek425ster It returns a dict. So, change this to something like: `dict: dict of type {node: xml}`